synapse/replication/http/login.py

```diff
@@ -91,8 +96,8 @@ async def _handle_request(self, request, user_id):
             auth_provider_session_id=auth_provider_session_id,
         )
 
-        return 200, res
+        return 200, cast(JsonDict, res)
```

Comment:
`register_device_inner` returns a `TypedDict`, so we just tell mypy to go away.

Reply:
FWIW I've been trying to leave a comment justifying every `cast` I make. But maybe TypedDict -> Dict is common enough that it's okay.